documentation/contributing/code/index.markdown

```diff
@@ -71,6 +71,22 @@ MoveIt uses *one* folder for all headers from all of its modules: ``include/move
 This is rather non-standard for catkin - catkin would prefer to have headers of each ROS package in a separate folder,
 e.g. `include/moveit_core/...`, `include/moveit_ros_planning/...`, etc.
 
+## pre-commit Formatting Checks
+
+In many of our repos we have a [pre-commit](https://pre-commit.com/) check that runs in CI.
+You can run this locally and in many cases it will fix formatting issues for you.
+To install use pip:
+
+    pip3 install pre-commit
+
+To run over all the files in the repo manually:
+
+    pre-commit run --all-files
+
+To install git hooks in your local repo:
+
+    pre-commit install
+
 ## clang-format Auto Code Formatting
 
 You can run **clang-format** in several ways. To install on Ubuntu simply run:
```

Comment:
As mentioned on Discord, I'd add "If you use pre-commit as described above, clang-format is applied automatically. This section describes how to use it manually."

Reply:
Done